pkg/aws/cloud.go

```diff
@@ -44,9 +47,28 @@ type Cloud interface {
 
 // NewCloud constructs new Cloud implementation.
 func NewCloud(cfg CloudConfig, metricsRegisterer prometheus.Registerer) (Cloud, error) {
+	hasIPv4 := true
+	addrs, err := net.InterfaceAddrs()
+	if err == nil {
+		hasIPv4 = false
+		for _, addr := range addrs {
+			str := addr.String()
+			if !strings.HasPrefix(str, "127.") && !strings.Contains(str, ":") {
```

Comment:
i think this check is a bit hacky.
maybe just upgrade the sdk and let users specify `AWS_EC2_METADATA_SERVICE_ENDPOINT_MODE` env variable via helm chart?

Reply:
Why should users have to specify AWS_EC2_METADATA_SERVICE_ENDPOINT_MODE at all?